### PR TITLE
IGNITE-14670 Update metastorage manager with: range with applied revision, invoke for multiple updates.

### PR DESCRIPTION
**Pull request by kgusakov**

### CODE REVIEW

File: modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/MetaStorageManager.java [hunk 1]
```diff
@@ -333,6 +342,27 @@ public synchronized void deployWatches() {
         return metaStorageSvc.range(keyFrom, keyTo, revUpperBound);
     }
 
+    /**
+     * Retrieves entries for the given key range in lexicographic order.
+     * Entries will be filtered out by the current applied revision as an upper bound
```
Comment (alievmirza) [inline modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/MetaStorageManager.java:347]: Probably we should explain here what applied revision is.

**Review by alievmirza — CHANGES_REQUESTED** [review]

File: modules/metastorage-common/src/main/java/org/apache/ignite/metastorage/common/Condition.java [hunk 2]
```diff
@@ -64,11 +64,16 @@ public boolean test(Entry e) {
          */
         private long rev;
 
+        /**
+         * Key of entry, which will be tested for condition.
```
Comment (sk0x50) [inline modules/metastorage-common/src/main/java/org/apache/ignite/metastorage/common/Condition.java:68]: It should be a one-line comment.

File: modules/metastorage-common/src/main/java/org/apache/ignite/metastorage/common/Condition.java [hunk 3]
```diff
@@ -254,11 +263,16 @@ public Condition le(long rev) {
          */
         private byte[] val;
 
+        /**
+         * Key of entry, which will be tested for condition.
```
Comment (sk0x50) [inline modules/metastorage-common/src/main/java/org/apache/ignite/metastorage/common/Condition.java:267]: Please use one-line Javadoc for such cases.

File: modules/metastorage-common/src/main/java/org/apache/ignite/metastorage/common/Conditions.java [hunk 4]
```diff
@@ -40,8 +53,12 @@
      * @return Condition on entry value.
      * @see Condition.ValueCondition
      */
-    public static Condition.ValueCondition value() {
-        return new Condition.ValueCondition();
+    public Condition.ValueCondition value() {
+        return new Condition.ValueCondition(key);
+    }
+
+    public static Conditions key(Key key) {
```
Comment (sk0x50) [inline modules/metastorage-common/src/main/java/org/apache/ignite/metastorage/common/Conditions.java:60]: Please add Javadoc here.

File: modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/MetaStorageManager.java [hunk 5]
```diff
@@ -303,15 +306,21 @@ public synchronized void deployWatches() {
...
+     * @see MetaStorageService#invoke(Condition, Collection, Collection)
      */
-    public @NotNull CompletableFuture<Boolean> invoke(
-        @NotNull Key key,
-        @NotNull Condition cond,
-        @NotNull Operation success,
-        @NotNull Operation failure
-    ) {
-        return metaStorageSvc.invoke(key, cond, success, failure);
+    public @NotNull CompletableFuture<Boolean> invoke(@NotNull Condition cond,
```
Comment (sk0x50) [inline modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/MetaStorageManager.java:313]: coding guideline does not allow that. please see https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines#CodingGuidelines-MethodArguments

I think it should be as follows:
```
    public @NotNull CompletableFuture<Boolean> invoke(
        @NotNull Condition cond,
        @NotNull Operation success,
        @NotNull Operation failure
    ) {
```

**Pull request closed without merging**